### PR TITLE
core: fix issue with git init

### LINKED ISSUE #8457: Husky Git hooks not installed on generated application
##### **Overview of the issue**

During the generation during the automatic npm install
>     husky > setting up git hooks
>     Can't find .git, skipping Git hooks installation.
>     Please check that you're in a cloned repository or run 'git init' to create an empty Git repository and reinstall husky.

##### **Motivation for or Use Case**

Git hooks are not installed, since husky is installed before git repository is initiated.

##### **Reproduce the error**

Generate a jhipster application with client (gateway or monolith).

##### **Suggest a Fix**

Run npm install after git init?

##### **JHipster Version(s)**

5.4.0

##### **JHipster configuration**

{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.4.1",
    "applicationType": "monolith",
    "baseName": "jhi_test_8457",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "NGRmMzgyZjVmOWJiOGFkOWJhZmM1OTQwOTg3YTRhNDE1NWIwMTA3NWI5OTUyMDIwZDExNzBlYmE5OTBjZDFkMTcxMTU4MGE0MDNiNDI5NTA2YjI0OTk5OTE1YTZiYTBhMDgxZDA4ZDdhOTE2MzRiMmE4MGJlMzk4MmEzZDkzMGU=",
    "clientFramework": "angularX",
    "useSass": true,
    "clientPackageManager": "npm",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en"
    ]
  }
}

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by deepu105**
fix #8457
-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -441,20 +460,15 @@ module.exports = class extends BaseGenerator {
...
+                        if (code === 0 && this.gitInitialized) {
+                            this.gitExec('add -A', { trace: false }, () => {
+                                this.gitExec(
+                                    `commit -am "Initial application generated by JHipster-${this.jhipsterVersion}"`,
+                                    { trace: false },
+                                    () => this.log(chalk.green.bold('Application successfully committed to Git.'))
+                                );
                             });
                         } else {
                             this.warning('The generated application could not be added to Git, as Git is not installed on your system');
```
Comment (murdos) [inline generators/app/index.js]: I was working on this issue too, and did nearly the same things locally.
But I've also split localInstall() (npm install + git), and moved npm install in the 'install' phase : see https://github.com/murdos/generator-jhipster/commit/0920fcb3f1ed47483b56bda048d3abbc04ef28cb

Comment (murdos) [inline generators/app/index.js]: This message could also be changed : "The generated application could not be added to Git, as Git repository is not initiated"

Comment (deepu105) [reply generators/app/index.js]: Npm install is in end phase instead of install for a reason (for blueprints and multiple app generation to work properly)

Comment (deepu105) [reply generators/app/index.js]: I'll update it later

Comment (murdos) [reply generators/app/index.js]: Ok :-)

**Comment by jdubois** [comment]
@deepu105  I changed the wording as I wanted to merge this, but no time to fix the conflict :-(

**Pull request merged**